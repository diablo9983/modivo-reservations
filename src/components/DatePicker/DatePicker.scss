
@mixin elo {
  content: "";
  display: block;
  position: absolute;
  background-color: #EDF5EC;
  width: 4px;
  height: 32px;
  top: -1px;
}

.datepicker {
  background-color: #fff;
  box-shadow: 0 10px 40px rgba(0, 0, 0, 0.15);
  border-radius: 20px;
  width: 300px;
  padding: 24px;

  &__header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    border: 1px solid #D8D8D8;
    border-radius: 48px;
    padding: 8px 10px;
    margin-bottom: 10px;
  }

  &__change-month {
    background-color: transparent;
    border: 0;
    width: 32px;
    height: 32px;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    border-radius: 50%;

    &:hover {
      background-color: #f0f0f0;
    }
  }

  &__current-date {
    font-size: 14px;
    font-weight: 700;
  }

  &__calendar {
    display: flex;
    flex-wrap: wrap;
    width: 252px;
    grid-gap: 4px;
    justify-content: space-between;
    text-align: center;
  }

  &__day {
    position: relative;

    &, &-symbol {
      width: 32px;
      height: 32px;
      font-size: 12px;
      font-weight: 600;
      border-radius: 50%;
      border: 1px solid transparent;
      display: flex;
      align-items: center;
      justify-content: center;
    }

    &--weekday {
      color: #999999;
    }

    &--inactive, &--disabled {
      color: #D8D8D8;
    }

    &--disabled {
      cursor: not-allowed;
    }

    &--today {
      color: #4E9845;
      border-color: #4E9845;
    }

    &:not(#{&}--weekday):not(#{&}--disabled):hover, &--in-range,
    &--start-range#{&}--connected, &--end-range#{&}--connected {
      background-color: #EDF5EC;
      color: #4E9845;
    }

    &:not(#{&}--weekday):not(#{&}--disabled):hover {
      cursor: pointer;
    }

    &--start-range:not(#{&}--connected), &--end-range:not(#{&}--connected),
    &--start-range &-symbol, &--end-range &-symbol {
      background-color: #4E9845;
      color: #fff;
    }

    &--start-range &-symbol,
    &--end-range &-symbol{
      position: absolute;
      left: -1px;
      top: -1px;
    }

    &--start-range#{&}--connected {
      border-top-right-radius: 0;
      border-bottom-right-radius: 0;
    }
    &--end-range#{&}--connected {
      border-top-left-radius: 0;
      border-bottom-left-radius: 0;
    }

    &--start-range#{&}--connected {
      &::after {
        @include elo();
        left: calc(100% + 1px);
      }
    }
    &--end-range#{&}--connected {
      &::before {
        @include elo();
        right: calc(100% + 1px);
      }
    }

    &--in-range {
      border-radius: 0;

      &:not(:nth-child(7n+1))::before, &:not(:nth-child(7n))::after {
        @include elo();
      }
      &::before {
        right: 100%;
      }
      &::after {
        left: 100%;
      }
    }
  }
}
