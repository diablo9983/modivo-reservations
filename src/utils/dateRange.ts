import dayjs from "dayjs";

export function dateRange(
	start: Date,
	end: Date
): Date[] {
	const startDate = dayjs(start);
	const endDate = dayjs(end);
	const diffInUnits = endDate.diff(startDate, "day") + 1;
	return Array.from(Array(diffInUnits).keys()).map((i) => {
		return startDate.add(i, "day").toDate()
	});
}
