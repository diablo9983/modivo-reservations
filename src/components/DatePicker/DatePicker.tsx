import {computed, defineComponent, ref} from "vue";
import type {PropType} from "vue";
import dayjs from "dayjs";
import weekday from "dayjs/plugin/weekday";
import localeData from "dayjs/plugin/localeData";
import isSameOrBefore from "dayjs/plugin/isSameOrBefore";
import isSameOrAfter from "dayjs/plugin/isSameOrAfter";
import isBetween from "dayjs/plugin/isBetween";
import "dayjs/locale/en";
import "./DatePicker.scss";
import clsx from "clsx";

dayjs.extend(weekday);
dayjs.extend(localeData);
dayjs.extend(isSameOrBefore);
dayjs.extend(isSameOrAfter);
dayjs.extend(isBetween);

dayjs.locale("en");

interface DateCellInterface {
    date: dayjs.Dayjs,
    dayOfMonth: number,
    isCurrentMonth: boolean
}

const daysInMonth = (year: string, month: string) => dayjs(`${year}-${month}-01`).daysInMonth();
const weekDays = dayjs.weekdaysShort();

function createDaysForCurrentMonth(year: string, month: string): DateCellInterface[] {
    return [...Array(daysInMonth(year, month))].map((day, index) => {
        return {
            date: dayjs(`${year}-${month}-${index + 1}`),
            dayOfMonth: index + 1,
            isCurrentMonth: true
        };
    });
}

function createDaysForPreviousMonth(year: string, month: string, currentMonthDays: DateCellInterface[]): DateCellInterface[] {
    const firstDayOfTheMonthWeekday = dayjs(currentMonthDays[0].date).weekday();

    const previousMonth = dayjs(`${year}-${month}-01`).subtract(1, "month");

    const previousMonthLastMondayDayOfMonth = dayjs(
        currentMonthDays[0].date
    ).subtract(firstDayOfTheMonthWeekday, "day").date();

    return [...Array(firstDayOfTheMonthWeekday)].map((day, index) => {
        return {
            date: dayjs(
                `${previousMonth.year()}-${previousMonth.month() + 1}-${previousMonthLastMondayDayOfMonth + index}`
            ),
            dayOfMonth: previousMonthLastMondayDayOfMonth + index,
            isCurrentMonth: false
        };
    });
}

function createDaysForNextMonth(year: string, month: string, currentMonthDays: DateCellInterface[]): DateCellInterface[] {
    const lastDayOfTheMonthWeekday = dayjs(`${year}-${month}-${currentMonthDays.length}`).weekday();

    const visibleNumberOfDaysFromNextMonth = lastDayOfTheMonthWeekday ? 6 - lastDayOfTheMonthWeekday : lastDayOfTheMonthWeekday

    return [...Array(visibleNumberOfDaysFromNextMonth)].map((day, index) => {
        return {
            date: dayjs(`${year}-${Number(month) + 1}-${index + 1}`),
            dayOfMonth: index + 1,
            isCurrentMonth: false
        }
    })
}

const getDays = (year: string, month: string): DateCellInterface[] => {
    const currentMonthDays = createDaysForCurrentMonth(year, month);
    const previousMonthDays = createDaysForPreviousMonth(year, month, currentMonthDays)
    const nextMonthDays = createDaysForNextMonth(year, month, currentMonthDays);

    return [...previousMonthDays, ...currentMonthDays, ...nextMonthDays];
}

function dateRange(
    start: Date,
    end: Date
): (number | Date)[] {
    const startDate = dayjs(start);
    const endDate = dayjs(end);
    const diffInUnits = endDate.diff(startDate, "day") + 1;
    return Array.from(Array(diffInUnits).keys()).map((i) => {
        return startDate.add(i, "day").toDate()
    });
}

export default defineComponent({
    name: "DatePicker",
    props: {
        type: {
            type: String as PropType<"from" | "to">,
            default: "start"
        },
        selectedStartDate: Date,
        selectedEndDate: Date,
        initialDate: Date,
        unavailableDates: Array as PropType<(Date | { from: Date, to: Date })[]>
    },
    emits: {
        select: (date: Date | null) => date instanceof Date || date === null
    },
    setup(props, { emit }) {
        const today = dayjs();

        const currentDate = ref(dayjs(props.initialDate || today));

        const disabledDates = computed(() => {
            return props.unavailableDates?.flatMap(date => {
                if (date instanceof Date) {
                    return dayjs(date)
                } else {
                    return dateRange(date.from, date.to).map(date => dayjs(date));
                }
            })
        });

        const days = computed(() => {
            return getDays(currentDate.value.format("YYYY"), currentDate.value.format("M"));
        });

        const showPrevMonth = () => {
            currentDate.value = dayjs(`${currentDate.value.format("YYYY")}-${currentDate.value.format("M")}-01`).subtract(1, "month");
        }
        const showNextMonth = () => {
            currentDate.value = dayjs(`${currentDate.value.format("YYYY")}-${currentDate.value.format("M")}-01`).add(1, "month");
        }
        const handleDateClick = (day: DateCellInterface) => {
            const startDate = props.type === "from" ? day.date.toDate() : props.selectedStartDate;
            const endDate = props.type === "to" ? day.date.toDate() : props.selectedEndDate;

            if (startDate && endDate) {
                const daysInRange = dateRange(startDate, endDate).map(date => dayjs(date).format("YYYY-MM-DD"));
                const disabled = (disabledDates.value || []).map(date => date.format("YYYY-MM-DD"));
                const includesDisabled = daysInRange.some(date => disabled.includes(date));

                if (includesDisabled) {
                    return;
                }
            }

            emit('select', day.date.toDate())
        }

        return {
            days,
            today,
            currentDate,
            disabledDates,
            handleDateClick,
            showPrevMonth,
            showNextMonth
        }
    },

    render() {
        return <div class={"datepicker"}>
            <div class="datepicker__header">
                <button class={"datepicker__change-month"} onClick={this.showPrevMonth}>
                    <svg width="8" height="14" viewBox="0 0 8 14" fill="none" xmlns="http://www.w3.org/2000/svg">
                        <path d="M6.7 13.405C6.60134 13.4063 6.50344 13.3875 6.41231 13.3497C6.32117 13.3118 6.23872 13.2558 6.17 13.185L0.955 7.97C0.698525 7.71232 0.554539 7.36356 0.554539 7C0.554539 6.63644 0.698525 6.28767 0.955 6.03L6.17 0.814997C6.23929 0.744852 6.32182 0.68916 6.4128 0.65115C6.50378 0.613139 6.6014 0.593567 6.7 0.593567C6.7986 0.593567 6.89622 0.613139 6.9872 0.65115C7.07818 0.68916 7.16071 0.744852 7.23 0.814997C7.36924 0.956261 7.4473 1.14665 7.4473 1.345C7.4473 1.54335 7.36924 1.73373 7.23 1.875L2.105 7L7.23 12.125C7.36924 12.2663 7.4473 12.4566 7.4473 12.655C7.4473 12.8533 7.36924 13.0437 7.23 13.185C7.16093 13.2553 7.07841 13.3111 6.98736 13.3489C6.89631 13.3867 6.79858 13.4057 6.7 13.405Z" fill="#333333"/>
                    </svg>
                </button>
                <span class={"datepicker__current-date"}>
                    {this.currentDate.format("MMMM")} {this.currentDate.format("YYYY")}
                </span>
                <button class={"datepicker__change-month"} onClick={this.showNextMonth}>
                    <svg width="8" height="14" viewBox="0 0 8 14" fill="none" xmlns="http://www.w3.org/2000/svg">
                        <path d="M1.3 13.405C1.20142 13.4057 1.10369 13.3867 1.01264 13.3489C0.921591 13.3111 0.839074 13.2553 0.77 13.185C0.63076 13.0437 0.552704 12.8533 0.552704 12.655C0.552704 12.4566 0.63076 12.2663 0.77 12.125L5.895 7L0.77 1.875C0.63076 1.73373 0.552704 1.54335 0.552704 1.345C0.552704 1.14665 0.63076 0.956261 0.77 0.814997C0.839294 0.744852 0.921823 0.68916 1.0128 0.65115C1.10378 0.613139 1.2014 0.593567 1.3 0.593567C1.3986 0.593567 1.49622 0.613139 1.5872 0.65115C1.67818 0.68916 1.76071 0.744852 1.83 0.814997L7.045 6.03C7.30148 6.28767 7.44546 6.63644 7.44546 7C7.44546 7.36356 7.30148 7.71232 7.045 7.97L1.83 13.185C1.76128 13.2558 1.67883 13.3118 1.58769 13.3497C1.49656 13.3875 1.39866 13.4063 1.3 13.405Z" fill="#333333"/>
                    </svg>
                </button>
            </div>

            <div class={"datepicker__calendar"}>
                {weekDays.map(weekday => (
                   <div class={"datepicker__day datepicker__day--weekday"}>
                       {weekday}
                   </div>
                ))}
                {this.days.map((day) => {
                    const disabled = (this.type === "from" && this.selectedEndDate && day.date.isSameOrAfter(this.selectedEndDate, "day"))
                        || (this.type === "to" && this.selectedStartDate && day.date.isSameOrBefore(this.selectedStartDate, "day"))
                        || !!this.disabledDates?.find((date: dayjs.Dayjs) => day.date.isSame(date));

                    const isToday = this.today.isSame(day.date, "date");
                    const isStartDate = day.date.isSame(this.selectedStartDate);
                    const isEndDate = day.date.isSame(this.selectedEndDate);
                    const inRange = this.selectedStartDate && this.selectedEndDate && day.date.isBetween(this.selectedStartDate, this.selectedEndDate);
                    const isConnected = (isStartDate || isEndDate) && this.selectedStartDate && this.selectedEndDate

                    return (
                        <div class={clsx(
                                "datepicker__day",
                                !day.isCurrentMonth && "datepicker__day--inactive",
                                isToday && "datepicker__day--today",
                                disabled && "datepicker__day--disabled",
                                isStartDate && "datepicker__day--start-range",
                                isEndDate && "datepicker__day--end-range",
                                inRange && "datepicker__day--in-range",
                                isConnected && "datepicker__day--connected"
                            )}
                            onClick={() => !disabled && this.handleDateClick(day)}
                        >
                            <span class={"datepicker__day-symbol"}>
                                {day.date.format("DD")}
                            </span>
                        </div>
                    )
                })}
            </div>
        </div>
    }
});
