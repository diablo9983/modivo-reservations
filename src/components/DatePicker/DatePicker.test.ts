import {describe, expect, test} from "vitest";
import {mount} from "@vue/test-utils";
import DatePicker from "@/components/DatePicker/DatePicker";
import type {DateRange} from "@/components/DatePicker/DatePicker";
import dayjs from "dayjs";
import {dateRange} from "@/utils/dateRange";

const dates = {
    today: new Date(),
    past: new Date("01 Jan 1970"),
    future: new Date("25 May 2031")
}

const validRange = [new Date("12 May 2022"), new Date("17 May 2022")];
// const invalidRange = [new Date("08 May 2022"), new Date("12 May 2022")];

const disabledDates: (Date | DateRange)[] = [
    new Date("10 May 2022"),
    {from: new Date("28 May 2022"), to: new Date("30 May 2022")}
];

describe("DatePicker", () => {
    describe(":props", () => {
        describe(':initialDate - render proper month based on initial date', () => {
            for (const date of Object.values(dates)) {
                test(date.toString(), async () => {
                    const picker = mount(DatePicker, {
                        props: {
                            initialDate: date
                        }
                    });

                    await picker.vm.$nextTick();

                    expect(picker.find('.datepicker__current-date').text()).toEqual(dayjs(date).format("MMMM YYYY"));
                });
            }
        });

        test(':selectedStartDate - properly shows selected start day as active', async () => {
            const picker = mount(DatePicker, {
                props: {
                    selectedStartDate: validRange[0]
                }
            });
            await picker.vm.$nextTick();

            expect(picker.find(`[data-date="${dayjs(validRange[0]).format("DD-MM-YYYY")}"]`).classes()).toContain("datepicker__day--start-range");
        });

        test(':selectedEndDate - properly shows selected end day as active', async () => {
            const picker = mount(DatePicker, {
                props: {
                    selectedEndDate: validRange[1]
                }
            });
            await picker.vm.$nextTick();

            expect(picker.find(`[data-date="${dayjs(validRange[1]).format("DD-MM-YYYY")}"]`).classes()).toContain("datepicker__day--end-range");
        });

        test(':unavailableDates - properly shows disabled date as disabled', async () => {
            const picker = mount(DatePicker, {
                props: {
                    unavailableDates: [disabledDates[0]]
                }
            });
            await picker.vm.$nextTick();

            expect(picker.find(`[data-date="${dayjs(disabledDates[0] as Date).format("DD-MM-YYYY")}"]`).classes()).toContain("datepicker__day--disabled");
        });

        test(':unavailableDates - properly shows disabled range of dates as disabled', async () => {
            const picker = mount(DatePicker, {
                props: {
                    unavailableDates: [disabledDates[1]]
                }
            });
            await picker.vm.$nextTick();

            const range = disabledDates[1] as DateRange;
            const dates = dateRange(range.from, range.to);

            for (const date of dates) {
                expect(picker.find(`[data-date="${dayjs(date).format("DD-MM-YYYY")}"]`).classes()).toContain("datepicker__day--disabled");
            }
        });
    });

    test('properly shows today day as active', async () => {
        const picker = mount(DatePicker);
        await picker.vm.$nextTick();

        expect(picker.find(`[data-date="${dayjs(dates.today).format("DD-MM-YYYY")}"]`).classes()).toContain("datepicker__day--today")
    });

    test("shows previous month on previous month button click", async () => {
        const picker = mount(DatePicker);
        await picker.vm.$nextTick();

        const element = picker.find(".datepicker__change-month--previous");
        await element.trigger("click");
        await picker.vm.$nextTick();

        expect(picker.find('.datepicker__current-date').text()).toEqual(dayjs().subtract(1, "month").format("MMMM YYYY"));
    });

    test("shows previous month on next month button click", async () => {
        const picker = mount(DatePicker);
        await picker.vm.$nextTick();

        const element = picker.find(".datepicker__change-month--next");
        await element.trigger("click");
        await picker.vm.$nextTick();

        expect(picker.find('.datepicker__current-date').text()).toEqual(dayjs().add(1, "month").format("MMMM YYYY"));
    });
})
