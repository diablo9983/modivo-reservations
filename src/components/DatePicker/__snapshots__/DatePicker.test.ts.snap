// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"datepicker\\">
  <div class=\\"datepicker__header\\"><button class=\\"datepicker__change-month\\"><svg width=\\"8\\" height=\\"14\\" viewBox=\\"0 0 8 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
        <path d=\\"M6.7 13.405C6.60134 13.4063 6.50344 13.3875 6.41231 13.3497C6.32117 13.3118 6.23872 13.2558 6.17 13.185L0.955 7.97C0.698525 7.71232 0.554539 7.36356 0.554539 7C0.554539 6.63644 0.698525 6.28767 0.955 6.03L6.17 0.814997C6.23929 0.744852 6.32182 0.68916 6.4128 0.65115C6.50378 0.613139 6.6014 0.593567 6.7 0.593567C6.7986 0.593567 6.89622 0.613139 6.9872 0.65115C7.07818 0.68916 7.16071 0.744852 7.23 0.814997C7.36924 0.956261 7.4473 1.14665 7.4473 1.345C7.4473 1.54335 7.36924 1.73373 7.23 1.875L2.105 7L7.23 12.125C7.36924 12.2663 7.4473 12.4566 7.4473 12.655C7.4473 12.8533 7.36924 13.0437 7.23 13.185C7.16093 13.2553 7.07841 13.3111 6.98736 13.3489C6.89631 13.3867 6.79858 13.4057 6.7 13.405Z\\" fill=\\"#333333\\"></path>
      </svg></button><span class=\\"datepicker__current-date\\">May 2022</span><button class=\\"datepicker__change-month\\"><svg width=\\"8\\" height=\\"14\\" viewBox=\\"0 0 8 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
        <path d=\\"M1.3 13.405C1.20142 13.4057 1.10369 13.3867 1.01264 13.3489C0.921591 13.3111 0.839074 13.2553 0.77 13.185C0.63076 13.0437 0.552704 12.8533 0.552704 12.655C0.552704 12.4566 0.63076 12.2663 0.77 12.125L5.895 7L0.77 1.875C0.63076 1.73373 0.552704 1.54335 0.552704 1.345C0.552704 1.14665 0.63076 0.956261 0.77 0.814997C0.839294 0.744852 0.921823 0.68916 1.0128 0.65115C1.10378 0.613139 1.2014 0.593567 1.3 0.593567C1.3986 0.593567 1.49622 0.613139 1.5872 0.65115C1.67818 0.68916 1.76071 0.744852 1.83 0.814997L7.045 6.03C7.30148 6.28767 7.44546 6.63644 7.44546 7C7.44546 7.36356 7.30148 7.71232 7.045 7.97L1.83 13.185C1.76128 13.2558 1.67883 13.3118 1.58769 13.3497C1.49656 13.3875 1.39866 13.4063 1.3 13.405Z\\" fill=\\"#333333\\"></path>
      </svg></button></div>
  <div class=\\"datepicker__calendar\\">
    <div class=\\"datepicker__day datepicker__day--weekday\\">Sun</div>
    <div class=\\"datepicker__day datepicker__day--weekday\\">Mon</div>
    <div class=\\"datepicker__day datepicker__day--weekday\\">Tue</div>
    <div class=\\"datepicker__day datepicker__day--weekday\\">Wed</div>
    <div class=\\"datepicker__day datepicker__day--weekday\\">Thu</div>
    <div class=\\"datepicker__day datepicker__day--weekday\\">Fri</div>
    <div class=\\"datepicker__day datepicker__day--weekday\\">Sat</div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">01</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">02</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">03</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">04</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">05</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">06</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">07</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">08</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">09</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">10</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">11</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">12</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">13</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">14</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">15</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">16</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">17</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">18</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">19</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">20</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">21</span></div>
    <div class=\\"datepicker__day datepicker__day--today\\"><span class=\\"datepicker__day-symbol\\">22</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">23</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">24</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">25</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">26</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">27</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">28</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">29</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">30</span></div>
    <div class=\\"datepicker__day\\"><span class=\\"datepicker__day-symbol\\">31</span></div>
    <div class=\\"datepicker__day datepicker__day--inactive\\"><span class=\\"datepicker__day-symbol\\">01</span></div>
    <div class=\\"datepicker__day datepicker__day--inactive\\"><span class=\\"datepicker__day-symbol\\">02</span></div>
    <div class=\\"datepicker__day datepicker__day--inactive\\"><span class=\\"datepicker__day-symbol\\">03</span></div>
    <div class=\\"datepicker__day datepicker__day--inactive\\"><span class=\\"datepicker__day-symbol\\">04</span></div>
  </div>
</div>"
`;
